delete.js

```diff
@@ -1,35 +1,42 @@
-const deleter = {
-  name_colunm_index: 0,
+class Deleter {
+  constructor(mainObject) {
+    this.name_colunm_index = 0;
+    this.allContacts = mainObject.addressBook;
+  }
 
   findContactToDelete(nameInput) {
-    let rows = addressBook.buildContactList();
+    let rows = this.allContacts.buildContactList();
     return rows.find(row => row.cells[this.name_colunm_index].textContent === nameInput);
-  },
+  }
 
   deleteContact(toBeDeleted) {
     toBeDeleted.remove();
   }
-};
+}
 
-const deletionUi = {
-  _userEntry: document.getElementById('user_contact_name'),
+class DeletionUi {
+  constructor(deleterObject, mainObject) {
+    this._userEntry = document.getElementById('user_contact_name');
+    this.eraser = deleterObject;
+    this.outputer = mainObject.mainUi;
+  }
```

Comment:
Again, you can name your arguments the same as the properties :+1: 

Also you can pass the objects directly. When you have a name so generic as `mainObject`, that should be a hint that you don't have a concept there.

Also, `mainUI` and `addressBook` are unrelated objects, so there is no need to group them in an object literal. They have nothing to do with each other, and the proof is that you are only using the `addressBook` in some objects (like the `Deleter`), and you are only using the `mainUI` in others (like the `DeletionUi`).

Only group stuff that is related or of the same type. Here you can just pass the objects directly. Also, I believe all your properties here can be private. :slightly_smiling_face: 

With all this in mind, we have:

```js
constructor(eraser, outputer) {
  this._userEntry = document.getElementById('user_contact_name');
  this._eraser = eraser;
  this._outputer = outputer;
}
```